plugins/ecc-vscc/ecc_validation_logic.go

```diff
@@ -121,70 +120,70 @@ func (vscc *VSCCECC) checkEnclaveEndorsement(cis *peer.ChaincodeInvocationSpec,
 		return err
 	}
 
-	for _, ns := range txRWSet.NsRwSets {
-		logger.Debugf("Namespace %s", ns.NameSpace)
-
-		// TODO make this more flexible
-		if !strings.HasPrefix(ns.NameSpace, "ecc") {
-			continue
-		}
+	function := cis.ChaincodeSpec.Input.Args[0]
+	logger.Debugf("function: %s\n", string(function))
 
-		function := cis.ChaincodeSpec.Input.Args[0]
-		logger.Debugf("function: %s\n", string(function))
-
-		if string(function) == "__setup" {
-			continue
-		}
+	if string(function) == "__setup" {
+		// do nothing for setup transaction as we handle this with ercc-vscc
```

Comment:
Just for clarifications -- overall, a few more comments for the whole plugin validations couldn't have hurt ... -- fabric calls validation plugins always on the (complete) transactions as as soon as there is a top-level or sub (cc2cc) transacation for a chaincode associated with a custom validation plugin?  I.e., for setup transaction both validation plugins are called -- and hence above works ... -- , any other (top-level) ercc transaction involves only ercc-vscc and any other (top-level) ecc tx involves only ecc-vscc? (regarding any other top-level ercc tx, i guess all would be rejected by the plugin with non-intuitve errors but would be safe, in any case, this plugin would be obsolete soon ...)

Reply:
I believe the `__setup` transaction will go away with ercc refactoring and the corresponding comments on how fabric deals with cc2cc invocations are then obsolete here.

I updated the comment but I refrain from adding details on how cc2cc are handled in fabric.